aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java

```diff
@@ -0,0 +1,459 @@
+import com.amplifyframework.auth.cognito.options.AWSCognitoAuthSignUpOptions;
+import com.amplifyframework.auth.result.AuthSignInResult;
+import com.amplifyframework.auth.result.AuthSignUpResult;
+import com.amplifyframework.auth.result.step.AuthNextSignInStep;
+import com.amplifyframework.auth.result.step.AuthNextSignUpStep;
+import com.amplifyframework.auth.result.step.AuthResetPasswordStep;
+import com.amplifyframework.auth.result.step.AuthSignInStep;
+import com.amplifyframework.auth.result.step.AuthSignUpStep;
+
+import com.amazonaws.mobile.client.AWSMobileClient;
+import com.amazonaws.mobile.client.Callback;
+import com.amazonaws.mobile.client.UserState;
+import com.amazonaws.mobile.client.UserStateDetails;
+import com.amazonaws.mobile.client.results.ForgotPasswordResult;
+import com.amazonaws.mobile.client.results.ForgotPasswordState;
+import com.amazonaws.mobile.client.results.SignInResult;
+import com.amazonaws.mobile.client.results.SignInState;
+import com.amazonaws.mobile.client.results.SignUpResult;
+import com.amazonaws.mobile.client.results.UserCodeDeliveryDetails;
+import com.amazonaws.mobile.config.AWSConfiguration;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.util.Map;
+
+import edu.emory.mathcs.backport.java.util.Collections;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Test that the current implementation of Auth as a wrapper of AWSMobileClient calls the correct
+ * AWSMobileClient methods with the correct parameters when the different Auth methods are called.
+ */
+@RunWith(RobolectricTestRunner.class)
+@SuppressWarnings("unchecked")
+public final class AuthComponentTest {
+    private AWSMobileClient mobileClient;
+    private AWSCognitoAuthPlugin authPlugin;
+    private final String userId = "myId";
+    private final String destination = "e***@email.com";
+    private final String deliveryMedium = "sms";
+    private final String attributeName = "email";
+    private final String username = "username";
+    private final String password = "password123";
+    private final String newPassword = "newPassword123";
+    private final String attributeKey = AuthUserAttributeKey.email().getKeyString();
+    private final String attributeVal = "email@email.com";
+    private final String confirmationCode = "confirm";
+    private final Map<String, String> metadata = Collections.singletonMap("aCustomKey", "aCustomVal");
```

Comment:
At least `private static final String SCREAMING_SNAKE_VAR` -- but, do you have some data models that you can put these in, so that there aren't just a heap of `String`? Like maybe an `new SomeCognitoType("username", "email@foo.tld");`, etc?

Lastly, the non-final fields should appear _after_ these constants (L76, L77.)

In unit tests, I prefer _not_ to store constants at the class level. This is because, often, the values are only used in individual tests. There are still a lot of good reasons to label your constants, though.

```
@Test(expected = FrickinMurdersItException.class)
public void testCookieMonsterWillPower() {
    final String temptation = "Cookie";
    CookieMonster monster = new CookieMonster();
    monster.temp(temptation);
    // Well, what did we expect, sigh.
}
```

Reply:
Metadata is an open ended map in a number of our methods which we decided to keep open because it's meant for sending custom data. Also, I put these at the class level because these same values are used throughout the various methods so figured it would be better to have them defined once rather than redefining the same dummy values in each test.